core/docker/container-test.sh

```diff
@@ -13,7 +14,7 @@ function test_trino_starts {
 
     local CONTAINER_NAME=$1
     local PLATFORM=$2
-    CONTAINER_ID=$(docker run -d --rm --platform ${PLATFORM} "${CONTAINER_NAME}")
+    CONTAINER_ID=$(docker run -d --platform ${PLATFORM} "${CONTAINER_NAME}")
```

Comment:
`--rm` is generally recommended for CI, so document why it's not here.

Actually why is it?

Reply:
Because in case of failure we want to be able to see the logs (see https://github.com/trinodb/trino/blob/2f86396b5cd7066eea7251f9beafb69af5b3ffa8/core/docker/container-test.sh#L25-L26)

Reply:
I can make it log unconditionally to keep everything simpler but this script gets used interactively as well.

Reply:
container-test.sh is a test. i wou;ldn't be concerned about interactive usage printing too much, why would it?

Reply:
Because IIRC that gets called by `build-remote.sh`/`build-local.sh` which is what maintainers use to deploy the docker images.